single_kernel_mongo/managers/mongodb_operator.py

```
@@ -699,10 +701,10 @@ def set_password(self, username: str, password: str | None = None) -> tuple[str,
             )
 
         secret_id = self.mongo_manager.set_user_password(user, new_password)
-        if user == BackupUser:
+        if user == BackupUser and self.workload.active():
```

We don't seem to manage the case where the workload is invalid and are silently ignoring it - yet returning the new username / password

---

The workload here is mongodb.
An issue that was happening is that PBM would try to start before mongodb was running, causing it to fail to start and put the charm in error.
If the workload is failing to update the password, it will raise an error (this is in the mongo manager) BEFORE the secret is set, and this error bubbles up to the event handler to fail the action.